pkg/services/validator/aws.go

```diff
@@ -745,30 +772,47 @@ func readAwsCredentials(c *components.AWSPluginConfig, tc *cfg.TaskConfig, k8sCl
 		c.SessionToken = string(secret.Data["AWS_SESSION_TOKEN"])
 	}
 
-	useSTS, err := prompts.ReadBool("Configure Credentials for STS", false)
+	return nil
+}
+
+func readDirectAwsCredentials(c *components.AWSPluginConfig) error {
+	api, err := aws.NewAPI(c.Validator.Auth, c.Validator.DefaultRegion)
 	if err != nil {
 		return err
 	}
-	if useSTS {
-		c.Validator.Auth.StsAuth = &vpawsapi.AwsSTSAuth{}
-		c.Validator.Auth.StsAuth.RoleArn, err = prompts.ReadText("AWS STS Role ARN", c.Validator.Auth.StsAuth.RoleArn, false, -1)
-		if err != nil {
-			return err
-		}
-		c.Validator.Auth.StsAuth.RoleSessionName, err = prompts.ReadText("AWS STS Session Name", c.Validator.Auth.StsAuth.RoleSessionName, false, -1)
-		if err != nil {
-			return err
-		}
-		duration := stsDurationSeconds
-		if c.Validator.Auth.StsAuth.DurationSeconds != 0 {
-			duration = intToStringDefault(c.Validator.Auth.StsAuth.DurationSeconds)
-		}
-		c.Validator.Auth.StsAuth.DurationSeconds, err = prompts.ReadInt("AWS STS Session Duration", duration, 900, 43200)
+	_, err = api.IAM.GetUser(context.TODO(), nil)
+	// auth toolchain is configured, skip prompting for credentials
+	if err == nil || !strings.Contains(err.Error(), awsNoCredsErr) {
+		return nil
```

Comment:
You can't do an early return here because then the env vars won't get set.

Reply:
See note in convo below

Reply:
You're right. My concern for AWS was that we would use creds from the config file (`c.Validator.Auth`) that aren't necessarily present on the local machine. But that's incorrect because we only care about `c.Validator.Auth` for the STS details.